test/test.ml

```diff
@@ -204,27 +205,30 @@ let t_with_pipe ~encoding () =
   let@ ic_server, oc_client = RPC.Util_pipe.with_pipe ~close_noerr:true () in
 
   let active = Switch.create () in
-  let@ runner = Moonpool.Ws_pool.with_ ~num_threads:4 () in
 
   let client : Client.t =
-    Client.create ~active ~runner ~timer ~encoding ~ic:ic_client ~oc:oc_client
-      ()
+    Client.create ~active ~timer ~encoding ~ic:ic_client ~oc:oc_client ()
   in
+
   let@ () =
     Fun.protect ~finally:(fun () ->
         let@ _sp = Trace.with_span ~__FILE__ ~__LINE__ "client.close" in
-        Client.close_and_join client;
-        Moonpool.Runner.shutdown runner)
+        Client.close_and_join client)
   in
 
   (* thread for server *)
-  let server : RPC.Server_for_client.t =
-    let encoding = RPC.Encoding.read_from_ic ic_server in
-    RPC.Server_for_client.create (* ~active *)
-      ~runner ~timer ~services ~encoding ~ic:ic_server ~oc:oc_server ()
+  let@ runner = Moonpool.Ws_pool.with_ ~num_threads:4 () in
```

Comment:
Good to see the tests growing, but I think we need more of them :-/ The stress test is already good, but I'd add another one with just random packets instead of meaningful messages, just in case. Also, a stress test with a single server and a large number of clients is a good idea I think.

Reply:
fair enough. I need to split the tests into more separate files I think.